torch/csrc/lazy/core/int128.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include <cstdint>
+#include <limits>
+
+#if defined(_MSC_VER) && (defined(_M_X64) || defined(__x86_64__))
+// For MSVC, couldn't find a byte order macro
+#elif (!defined(__BYTE_ORDER__) || !defined(__ORDER_LITTLE_ENDIAN__) || \
+     __BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__)
+#error "Only little endian supported"
+#endif
+
+namespace torch {
+namespace lazy {
+
+class alignas(unsigned __int128) uint128 {
```

Comment:
Out of curiosity why not use `absl::` version of int128 utils and type?

Reply:
Yeah, definitely something to consider. The int128 here was taken straight from XLA, right? It doesn't seem like a very good implementation; probably should consider cribbing a more fully featured one from somewhere.

Reply:
I think this version was made (simplified) by @asuhan during the process of making lazy_tensor_core separate from XLA (and absl) dependencies.  What would it entail for PyTorch to use the absl 128 type? Is it a light enough dep that we would just add entire absl as a thirdparty dep to PyTorch?  Or are you proposing to copy just int128 from absl, and thereby get a more full featured one.

Reply:
copy it in. Also not sure if absl is the best lib to get 128-bit from; it might be a pass if it's very intricately tied to absl

Reply:
Apart from it being a 1K LOC file and needing relatively sparse changes to remove ABSL stuff, it has other `.inc` files I'd have to sort out which define int128 intrinsics, which I'm not sure we'd even use?
https://github.com/abseil/abseil-cpp/blob/master/absl/numeric/int128.h

```
// TODO(absl-team): This module is inconsistent as many inline `uint128` methods
// are defined in this file, while many inline `int128` methods are defined in
// the `int128_*_intrinsic.inc` files.
```

I could work on copying the absl 128 in if that's the preferred route.  I saw the chromium project also using absl's 128 and didn't find a lot of alternatives,

This could work too, but it's not a ton smaller than absl and I'd trust absl over it.
https://github.com/calccrypto/uint128_t/blob/master/uint128_t.include

Reply:
ok, so leave the basic int128 in lazy/core, or still move to c10/util? @ezyang

Reply:
up to you. It's still pretty reasonable to put it in c10/util on principle, but I guess if we think this implementation is crappy keeping it outside can prevent other people from starting to use it. The real problem is if someone else adds an int128 implementation without telling you